### PR TITLE
io.github.vemonet.EmojiMart

### PR DESCRIPTION
**Pull request by vemonet**
Submission of Emoji Mart, an emoji picker desktop app based on the emoji-mart web component, built with tauri

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/vemonet/EmojiMart**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: io.github.vemonet.EmojiMart.desktop
Comment (hfiguiere) [inline io.github.vemonet.EmojiMart.desktop:1]: this file should be upstream

File: io.github.vemonet.EmojiMart.appdata.xml
Comment (hfiguiere) [inline io.github.vemonet.EmojiMart.appdata.xml:1]: this file should be upstream

**Comment by vemonet** [comment]
bot, build io.github.vemonet.EmojiMart

**Comment by vemonet** [comment]
bot, build io.github.vemonet.EmojiMart

**Comment by vemonet** [comment]
bot, build io.github.vemonet.EmojiMart

**Comment by vemonet** [comment]
bot, build io.github.vemonet.EmojiMart

**Comment by vemonet** [comment]
bot, build io.github.vemonet.EmojiMart

**Comment by vemonet** [comment]
It seems that the `npm ci --offline` command cannot find the cached sources

The `node-sources.json` looks fine, and I defined the `npm_config_cache` as done by other successful builds:
* https://github.com/flathub/in.cinny.Cinny/blob/master/in.cinny.Cinny.yml
* https://github.com/hadess/flathub/blob/d4b53ff829e0917c5129294132f619e5f12d337c/io.github.neil_morrison44.pocket-sync.yaml

```yml
build-options:
      env:
        CARGO_HOME: /run/build/EmojiMart/cargo
        XDG_CACHE_HOME: /run/build/EmojiMart/flatpak-node/cache
        npm_config_cache: /run/build/EmojiMart/flatpak-node/npm-cache
        npm_config_offline: 'true'
```

So I am not what is going on here, any idea @hfiguiere ? Maybe I am missing something when defining the cache paths

**Comment by vemonet** [comment]
bot, build io.github.vemonet.EmojiMart

**Comment by vemonet** [comment]
The issue was coming from `flatpak-node-generator` missing some sources when generating from the `package-lock.json` (it generated sources for ~60 packages on ~100 needed)

I switched to yarn following this advice: https://github.com/flatpak/flatpak-builder-tools/issues/244

Unfortunately again `flatpak-node-generator` does not support `yarn.lock` files generated by yarn v3. So I needed to downgrade back to yarn v1

But it seems to work now!

**Comment by vemonet** [comment]
bot, build io.github.vemonet.EmojiMart

**Comment by vemonet** [comment]
bot, build io.github.vemonet.EmojiMart

**Comment by vemonet** [comment]
Working well, thanks!

**Comment by barthalion** [comment]
What should have happened here if GitHub API wasn't so wonky this week:

```
A repository for this submission has been created: https://github.com/flathub/io.github.vemonet.EmojiMart

You will receive an invitation to be a collaborator which will grant you write access to the repository above. The invite can be also viewed [here](https://github.com/flathub/io.github.vemonet.EmojiMart/invitations).

If you have never maintained an application before, common questions are answered in [the app maintenance guide](https://github.com/flathub/flathub/wiki/App-Maintenance).

Thanks!
```

**Pull request closed without merging**